src/map.cpp

```diff
@@ -7455,26 +7454,25 @@ void map::spawn_monsters_submap_group( const tripoint &gp, mongroup &group, bool
     }
 
     // Find horde's target submap
-    // TODO: fix point types
-    tripoint horde_target( tripoint( -abs_sub.xy(), abs_sub.z ) + group.target.xy().raw() );
-    sm_to_ms( horde_target );
+    tripoint_abs_sm horde_target = submap_pos + ( group.target - group.pos );
     for( auto &tmp : group.monsters ) {
         for( int tries = 0; tries < 10 && !locations.empty(); tries++ ) {
-            const tripoint p = random_entry_removed( locations );
-            if( !tmp.can_move_to( p ) ) {
+            const tripoint local_pos = random_entry_removed( locations );
+            const tripoint_abs_ms abs_pos = get_map().getglobal( local_pos );
+            if( !tmp.can_move_to( local_pos ) ) {
                 continue; // target can not contain the monster
             }
             if( group.horde ) {
                 // Give monster a random point near horde's expected destination
-                const tripoint rand_dest = horde_target +
-                                           point( rng( 0, SEEX ), rng( 0, SEEY ) );
-                const int turns = rl_dist( p, rand_dest ) + group.interest;
+                const tripoint_abs_ms rand_dest = project_to<coords::ms>( horde_target ) + point( rng( 0, SEEX ),
+                                                  rng( 0, SEEY ) );
```

Comment:
This line has grown to be rather ugly.  Consider e.g.
```suggestion
                point_sm_ms pos_in_sm( rng( 0, SEEX ), rng( 0, SEEY ) );
                const tripoint_abs_ms rand_dest = project_combine( horde_target, pos_in_sm );
```

Reply:
Done (and you forgot const :)

Reply:
Hmm, it also needs to be `point_rel_ms`, `point_sm_ms` doesn't work.

Reply:
If you're just adding them then you'd need `point_rel_ms`m but I thought `point_sm_ms` would work with `project_combine`?